app/controllers/recipes_controller.rb

```diff
@@ -3,18 +3,31 @@ class RecipesController < ApplicationController
   before_action :authenticate_user!, except: %i[index show]
 
   def index
-    @recipes = Recipe.includes(:user).all
+    @recipes = Recipe.includes([:user]).where(user: current_user)
   end
 
+  # def show
+  #   @recipe = Recipe.includes(:user, :recipe_foods).find(params[:id])
+  #   @recipe_food = RecipeFood.new
+
+  #   if @recipe.public? || current_user == @recipe.user
+  #     # Display the recipe details as in the wireframe
+  #   else
+  #     redirect_to recipes_path, alert: 'You are not authorized to view this private recipe.'
+  #   end
+  # end
+
   def show
-    @recipe = Recipe.includes(:user, :recipe_foods).find(params[:id])
-    @recipe_food = RecipeFood.new
+    @recipe = Recipe.find(params[:id])
```

Comment:
The N+1 problem in the method was already fixed. kindly replace this line with the previous one `@recipe = Recipe.includes(:user, :recipe_foods).find(params[:id])`